### PR TITLE
AI will now usually choose not to eat what it can't digest

### LINKED ISSUE #4440: Allow AI to consider digestability when scanning for potential prey.
AI predators currently do not consider if they can digest their prey or not when scanning for suitable prey in the area. This leads to predators repeatedly attempting to consume prey that they cannot digest to no avail.

When scanning for prey, AI should have a chance to consider if they can digest it or not (By taking into account cell walls and presence of pre-requisite lysosome enzymes) based off of their behavioral scales. Brave and/or opportunistic AI should be more likely to attempt to consume undigestable prey items.

### PR DESCRIPTION
**Pull request by adQuid**
**Brief Description of What This PR Does**

Adds a rolled chance on target acquisition to probably not engulf things it can't digest.

**Related Issues**

fixes #4440 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: simulation_parameters/Constants.cs [hunk 1]
```diff
@@ -740,6 +740,9 @@ public static class Constants
     // Cooldown for AI for toggling engulfing
     public const float AI_ENGULF_INTERVAL = 300;
 
+    // Probability, rolled at each AI step (which very often), that the AI will try to engulf something it can't eat
+    public const float AI_BAD_ENGULF_CHANCE = 0.25f;
```
Comment (hhyyrylainen) [inline simulation_parameters/Constants.cs:744]: Even though other people have commented incorrectly the AI constants, could you make this into a proper XML comment?

**Review by buckly90 — APPROVED** [review]
This works just as well as I hoped, great stuff!

I would say that the value for AI_BAD_ENGULF_CHANCE feels a bit too big. Somewhere around 10-15% might be better.

I noticed that cells that chose to attempt engulfing something they couldn't digest would begin to use toxins as well. I assume this behavior normally wouldn't be an issue, but with digestability now part of the equation, hunting with toxins versus engulfing might need to be seperated...
While you are free to try fixing that in this PR, I'll otherwise go ahead and open an issue about this once this PR is merged.

**Comment by hhyyrylainen** [comment]
I think we ought to leave out the more complex change for later, and open an issue about it to not forget about it.

@adQuid is there a reason why you haven't checked the PR workflow first item (`PR author has checked that this PR works as intended and doesn't--`)?

File: simulation_parameters/Constants.cs [hunk 2]
```diff
@@ -740,6 +740,11 @@ public static class Constants
     // Cooldown for AI for toggling engulfing
     public const float AI_ENGULF_INTERVAL = 300;
 
+    /// <summary>
+    ///   Probability, rolled at each AI step (which very often), that the AI will try to engulf something it can't eat
```
Comment (hhyyrylainen) [inline simulation_parameters/Constants.cs:744]: I'm still seeing the "(which very often)" part which doesn't sound grammatically correct to me here.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
The eating decision logic is pretty complex but as far as I can parse it in my head it is correct. I noticed one code comment typo, well at least I think it is one, that didn't get fixed. If that's fixed then with buckly's testing I think this is fine to merge.

File: simulation_parameters/Constants.cs (hunk 2 above)
Comment (hhyyrylainen) [reply simulation_parameters/Constants.cs]: And now the line is too long :laughing:

**Review by hhyyrylainen — APPROVED** [review]

**Comment by hhyyrylainen** [comment]
@buckly90 do you want to open that follow up issue or should I?

**Comment by hhyyrylainen** [comment]
I opened the follow up issue so that I can move on: https://github.com/Revolutionary-Games/Thrive/issues/4445 let me know if I understood incorrectly what it should have.

**Pull request merged**